README.md

```diff
@@ -55,6 +55,55 @@ test.all(common)
 
 A valid (read: that follows this interface) IPFS core implementation, must expose the following API.
 
+## Files
+
+### `add`
+
+> Add files and data to IPFS.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.files.add(data, [callback])
+
+Where `data` may be
+
+- an array of objects, each of the form
+```js
+{
+  path: '/tmp/myfile.txt',
+  stream: fs.createReadStream('/tmp/myfile.txt')
+}
+```
+- an array of objects, each of the form
+```js
+{
+  path: '/tmp/myfile.txt',
+  content: new Buffer('hello')
+}
+```
+- a `Buffer` instance
+- a readable stream
+
+If no value `data` is provided, a Readable stream will be returned, which
+tuples like the above two object formats can be written.
+
+`callback` must follow `function (err, res) {}` signature, where `err` is an
+error if the operation was not successful and `res` is an array of objects of
+the form
+
+```js
+// js-ipfs-api
+{
+  Name: 'some name',
+  Hash: 'QmRNjDeKStKGTQXnJ2NFqeQ9oW23WcpbmvCVrpDHgDg3T6'
+}
+```
```

Comment:
Yes, I would like DAGNodes returned, but doesn't it come at a high cost? I.e. Won't doing an extra `object.get` call per added node mean N extra API requests against Core (in js-ipfs-api land, N HTTP requests)? Or is there some sort of batching `object.get` mechanism I don't know about?